### DISCUSSION TITLE
Best Practice: Looping and Modification

### DISCUSSION THREAD

**Discussion by JeremyVyska** [microsoft/alguidelines#52] [BC Best Practices]
A common operation is to want to set filters on a record, then loop through the resulting records to modify them.

We should probably have a Best Practices page that breaks down:

- Modify vs ModifyAll
- Usage of FINDSET(TRUE,TRUE)
- FINDSET with a 2nd variable for modification

Considerations and when to use each strategy.

**Comment by JeremyVyska** [comment]
@JesperSchulz - this might also be great for Kennie if he's up for it.

**Comment by git-no** [comment]
Looping is often a dedicated topic. The methods and commands in loop (especially Find)  depending on the goal and table methods (read, insert, modify, delete).
Insert has a *special* behaviour on performance, e.g. bulk insert and table locking.
Maybe it is an approach not to focus on the command (Modify or ModifyAll), than on the goal to achieve and which method is the best for it.
